### PR TITLE
feat: morpho blue protocol plugin 

### PR DESCRIPTION
**Pull request by robercano**
- Added Morpho protocol plugin + tests
- Reworked Lending Pool data type to contain debt and collateral token as part of the pool itself, and not only in the ID
- Some lending pool IDs like Aave, Spark and Maker still contain the debt/collateral token in the ID, for Morpho we use the market ID
- Reworked all common data types to use Zod schema for data definition and removed the schema checker as it is not needed anymore
- Type guards for common data types now indicate that the passed candidate is an instance of the type, instead of just the data
- Added Address Book manager to handle contract addresses requests
- Added `IManagerWithProviders` and `IManagerProvider` plus related base classes (`ManagerWithProvidersBase` and `ManagerProviderBase`) to consolidate common functionality for the managers
- Updated all tests accordingly
- Added comments and documentation

### CODE REVIEW

File: sdk/oracle-service/tests/mocks/MockOracleProvider.ts [hunk 1]
```diff
@@ -8,9 +9,18 @@ import {
...
+{
+  constructor() {
+    super({
+      configProvider: undefined as unknown as IConfigurationProvider,
+      type: OracleProviderType.OneInch,
+    })
+  }
 
   getSupportedChainIds(): ChainId[] {
     return [1, 4, 6]
```
Comment (paszkowskiDamian) [inline sdk/oracle-service/tests/mocks/MockOracleProvider.ts:26]: Don't we have enum with those id's? I'm aware that it wasn't added in this PR

Comment (robercano) [reply sdk/oracle-service/tests/mocks/MockOracleProvider.ts:26]: We have the `ChainFamiliesMap` where those IDs are defined, but not in an enum. In any case this is a Mock, not sure if we want to make it better. Let me know and I can change that.

File: sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+      .toBN()
+      .div(marketCollateralPriceInDebt.toBN().multipliedBy(marketInfo.totalSupplyAssets.toBN()))
+      .multipliedBy(100)
+
+    return RiskRatio.createFrom({
+      ratio: Percentage.createFrom({
+        value: riskRatioValue.toNumber(),
+      }),
+      type: RiskRatioType.LTV,
+    })
```
Comment (paszkowskiDamian) [inline sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts:396]: As far as I remember liquidation threshold for a pool in morpho is just `lltv` that you get from chain. The thing that you calculated is `LTV` of a pool.

File: sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+    const { morphoLendingPool, marketInfo, marketCollateralPriceInDebt } = params
+
+    const debtToken = morphoLendingPool.debtToken
+    const priceUSD = await this.ctx.oracleManager.getSpotPrice({
+      baseToken: debtToken,
+    })
+
+    const debtCeiling = marketCollateralPriceInDebt
+      .multiply(marketInfo.totalSupplyAssets)
+      .multiply(morphoLendingPool.lltv)
```
Comment (paszkowskiDamian) [inline sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts:219]: I think we might need to subtract current total pool debt, but I'm not sure, it depends how we use it.

File: sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+      .div(new BigNumber(10).pow(MorphoLLTVPrecision))
+      .multipliedBy(100)
+      .toNumber()
+
+    return {
+      debtToken,
+      collateralToken,
+      oracle: Address.createFromEthereum({ value: marketParameters[2] }),
+      irm: Address.createFromEthereum({ value: marketParameters[3] }),
+      lltv: Percentage.createFrom({ value: lltv }),
```
[the remark this answers is not in the document]
Comment (robercano) [reply sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts:371]: Changed it to RiskRatio with LTV type

File: sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts (hunk 2 above)
Comment (robercano) [reply sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts:396]: You are right. This should just be the Pool LLTV. I'll change it

**Pull request merged**